mm2src/mm2_main/src/lp_price.rs

```diff
@@ -0,0 +1,342 @@
+                        Some(base_price_infos.last_updated_timestamp)
+                    } else {
+                        Some(rel_price_infos.last_updated_timestamp)
+                    };
+                rate_infos.base_price = base_price_infos.last_price.clone();
+                rate_infos.rel_price = rel_price_infos.last_price.clone();
+                rate_infos.price = &base_price_infos.last_price / &rel_price_infos.last_price;
+                Some(rate_infos)
+            },
+            None => None,
+        }
+    }
+}
+
+#[cfg(not(target_arch = "wasm32"))]
+pub async fn process_price_request(price_url: &str) -> Result<TickerInfosRegistry, MmError<PriceServiceRequestError>> {
+    debug!("Fetching price from: {}", price_url);
+    let (status, headers, body) = mm2_net::native_http::slurp_url(price_url).await?;
+    let (status_code, body, _) = (status, std::str::from_utf8(&body)?.trim().into(), headers);
+    if status_code != StatusCode::OK {
+        return MmError::err(PriceServiceRequestError::HttpProcessError(body));
+    }
+    let model: HashMap<String, TickerInfos> = serde_json::from_str(&body)?;
+    Ok(TickerInfosRegistry(model))
+}
+
+#[cfg(target_arch = "wasm32")]
+pub async fn process_price_request(price_url: &str) -> Result<TickerInfosRegistry, MmError<PriceServiceRequestError>> {
+    debug!("Fetching price from: {}", price_url);
+    let (status, headers, body) = mm2_net::wasm_http::slurp_url(price_url).await?;
+    let (status_code, body, _) = (status, std::str::from_utf8(&body)?.trim().into(), headers);
+    if status_code != StatusCode::OK {
+        return MmError::err(PriceServiceRequestError::HttpProcessError(body));
+    }
+    let model: HashMap<String, TickerInfos> = serde_json::from_str(&body)?;
+    Ok(TickerInfosRegistry(model))
+}
+
+pub async fn fetch_price_tickers(price_url: &str) -> Result<TickerInfosRegistry, MmError<PriceServiceRequestError>> {
+    let model = process_price_request(price_url).await?;
+    debug!("price registry size: {}", model.0.len());
+    Ok(model)
+}
+
+#[derive(Default, Clone, Debug, PartialEq)]
+pub struct CEXRates {
+    pub base: BigDecimal,
+    pub rel: BigDecimal,
+}
+
+/// fetcher function to fetch latest price from a single endpoint.
+async fn try_price_fetcher_endpoint(
+    endpoint: &str,
+    base: &str,
+    rel: &str,
+) -> Result<CEXRates, MmError<PriceServiceRequestError>> {
+    let response = process_price_request(endpoint).await?;
+    let fiat_price = response
+        .get_cex_rates(base, rel)
+        .or_mm_err(|| PriceServiceRequestError::Internal("couldn't fetch price".to_string()))?;
+    let (base_usd_price, rel_usd_price) = fiat_price.get_rate_price();
+    Ok(CEXRates {
+        base: base_usd_price,
+        rel: rel_usd_price,
+    })
+}
+
+/// consume try_price_fetcher_endpoint result here using different endpoints.
+/// return price data on Success or None on failure.
```

Comment:
Please same here.
Also when you want to point to a function/structure/type etc, it's preferred to use `` characters.